src/database.py

```diff
@@ -0,0 +1,105 @@
+import sqlite3
+from datetime import datetime
+
+class Database:
+    def __init__(self, db_path='data/database/database.db'):
+        self.conn = sqlite3.connect(db_path)
+        self.cursor = self.conn.cursor()
+        self.create_tables()
+
+    def create_tables(self):
+        self.cursor.execute('''
+            CREATE TABLE IF NOT EXISTS interactions (
+                id INTEGER PRIMARY KEY,
+                npc_name TEXT,
+                interaction_time TIMESTAMP
+            )
+        ''')
+        self.cursor.execute('''
+            CREATE TABLE IF NOT EXISTS map_switches (
+                id INTEGER PRIMARY KEY,
+                from_map TEXT,
+                to_map TEXT,
+                switch_time TIMESTAMP
+            )
+        ''')
+        self.cursor.execute('''
+            CREATE TABLE IF NOT EXISTS goggles_wearing (
+                id INTEGER PRIMARY KEY,
+                worn BOOLEAN,
+                wear_time TIMESTAMP
+            )
+        ''')
+        self.cursor.execute('''
+            CREATE TABLE IF NOT EXISTS bath_house_entries (
+                id INTEGER PRIMARY KEY,
+                health_state TEXT,
+                health INTEGER,
+                entry_time TIMESTAMP
+            )
+        ''')
+        self.cursor.execute('''
+            CREATE TABLE IF NOT EXISTS group_switches (
+                id INTEGER PRIMARY KEY,
+                old_group TEXT,
+                new_group TEXT,
```

Comment:
The group is stored as an IntEnum in-game, so you will need to replace this by integers. Check `StudyGroup` inside `src.enums`

Reply:
you're right

Reply:
It’s a base class. I’m sure Sophie would like to add more data variables to save